spec/src/main/asciidoc/architecture.asciidoc

```diff
@@ -17,37 +17,144 @@
+be stored.
+([Examples](http://opentracing.io/documentation/pages/supported-tracers.html)).
+The storage service for distributed trace records can provide features to view
+the cross service trace records associated with particular request flows.
 
-Distributed tracing allows you to trace the flow of a request across service boundaries.
-This is particularly important in a microservices environment where a request typically flows through multiple services.
-To accomplish distributed tracing, each service must be instrumented to log messages with a correlation id that may have been propagated from an upstream service.
-A common companion to distributed trace logging is a service where the distributed trace records can be stored. ([Examples](http://opentracing.io/documentation/pages/supported-tracers.html)).
-The storage service for distributed trace records can provide features to view the cross service trace records associated with particular request flows.
+Opentracing.io is an API specification for working with a distributed tracing
+system. Opentracing.io defines a Span as the conceptual artifact that is
+propagated between services. The Span information can also be communicated to a
+central server for storage.
 
-Opentracing.io is an API specification for working with a distributed tracing system. Opentacing.io defines a Span as the conceptual artifact that is propagated between services. The Span information can also be communicated to a central server for storage.
+The microprofile.io Opentracing proposal will describe how a developer can
+easily specify methods to be traced.
 
-The microprofile.io OpenTracing proposal will describe how a developer can easily specify methods to be traced. The proposal will also describe when microprofile.io implementations can assist with Span propagation.
+The proposal will also describe how a microprofile.io implementation will assist with Span creation and propagation. It is particularly useful to be able to enable distributed tracing for an application without having to add any code at all to the application.
 
-Two annotations are provided to define how Spans can be used.
+=== Enabling distributed tracing with no code instrumentation
 
-* Trace: Specify a class or method to be traced.
+The microprofile.io implementation will allow JAX-RS applications to participate in distributed tracing, without requiring developers to add any distributed tracing code into their applications, and without requiring developers to know anything about the distributed tracing environment that their JAX-RS application will be deployed into.
 
-* NoTrace: Specify that a method is not to be traced.
+1. The microprofile.io implementation must provide a mechanism to configure an io.opentracing.Tracer implementation for use by each JAX-RS application.
+2. The microprofile.io implementation must provide a mechanism to automatically extract Span information from any incoming JAX-RS request.
```

Comment:
This is usually called Context or SpanContext information

Reply:
Added more info to spec